Assets/Scripts/Calendar UI.cs

```diff
@@ -80,46 +102,114 @@ private void HandleBackwardClick()
+    }
+    private void HandleSelectCell(Cell cell)
+    {
+        if (pop == null)
+        {
+
+            selectedCell = cell as DayCell;
+
+            pop = Instantiate(popUpPrefab, canvas);
+            pop.MakeButton("Sticker", HandleAddStickerPressed, false);
+            pop.MakeButton("Notes", HandleAddDescriptionPressed, false);
+            pop.SetPosition(cell.transform.position);
+
+
+        }
+    }
+    private void HandleAddDescriptionPressed()
+    {
+        EventManager.Instance.TriggerAddDescriptionPressed();
+        //Make function to trigger description scene;
+    }
+
 
-    public void PopulateCalendarGrid()
+    private void HandleAddStickerPressed()
     {
 
+        pop.DestroyButonCells();
+        pop.MakeButton("Poop", HandleAddPoopPressed, selectedCell.DData.isPoopImageActive);
+        pop.MakeButton("Pill", HandleAddMedicinePressed, selectedCell.DData.isMedicineImageActive);
+
+
+    }
+
+    private void HandleDESelectAllCells()
+    {
+        selectedCell.DeSelect();
+        selectedCell = null;
+        Destroy(pop.gameObject);
+        pop = null;
+    }
+
+    private void HandleAddPoopPressed()
+    {
+        selectedCell.SetPoopImage();
+
+
+    }
+
+    private void HandleAddMedicinePressed()
+    {
+        selectedCell.SetMedicineImage();
+    }
+    // _____________________________________________________________________________________
+    //   CLASS - SPECIEFIC FUNCTIONS
+    // _____________________________________________________________________________________
+    private void PopulateCalendarGrid()
+    {
+        ClearPopulationList();
         int totalDays = numberOfBlanksBefore + daysInMonth + numberOfBlanksAfter;
-        List<Cell> populationList = calendarGridPopulator.Populate(totalDays);
+        populationList = calendarGridPopulator.Populate(totalDays);
+
         int day = 1;
         for (int i = 0; i <= populationList.Count - 1; i++)
         {
-
+            string text;
+            Color color;
 
             if (i < numberOfBlanksBefore || i >= numberOfBlanksBefore + daysInMonth)
             {
-                populationList[i].SetImageColor(blankColor);
-                populationList[i].SetTextValue("");
+
+                text = "";
+                color = blankColor;
 
             }
 
             else
             {
 
-                populationList[i].SetTextValue(day.ToString());
+                text = day.ToString();
                 if (day == currentDate.Day && defaultDate.Month == currentDate.Month && defaultDate.Year == currentDate.Year)
                 {
-                    populationList[i].SetImageColor(currentDayColor);
+                    color = currentDayColor;
                 }
                 else
                 {
-                    populationList[i].SetImageColor(daysDefaultColor);
+                    color = daysDefaultColor;
                 }
                 day++;
             }
+            DayCellData data = new DayCellData();
```

Comment:
is a constructor with 2 parameters worth it here?

Reply:
Resolved